src/python/pants/option/global_options.py

```diff
@@ -984,7 +984,7 @@ class BootstrapOptions:
     )
     native_options_validation = EnumOption(
         default=NativeOptionsValidation.warning,
-        removal_version="2.26.0.dev0",
+        removal_version="2.28.0.dev0",
```

Comment:
I think we can remove this now since it's not consulted, and was only introduced recently as an escape hatch, so it's not something most people will have ever set.

Reply:
There is value in reducing codebase clutter...

Reply:
My thinking for postponing it is to ensure that someone on 2.24 who has to set it can easily validate future version (both 2.25.x RCs, and 2.26.x dev releases) without having to edit `pants.toml` for tiny things like "remove an irrelevant option". That is, try to keep it as smooth as possible for people to test pre-releases so we get more feedback.

I agree removing codebase clutter is nice, but this seems like such a small amount of clutter (a handful of declarative lines in a 2.3kloc file) that there's not much benefit in doing it too eagerly.

Reply:
Happy to push it less far forward, though:

```suggestion
        removal_version="2.27.0.dev0",
```